lib/mayaUsd/fileio/translators/translatorMayaReference.cpp

```diff
@@ -457,36 +518,56 @@ MStatus UsdMayaTranslatorMayaReference::update(const UsdPrim& prim, MObject pare
                             prim.GetPath().GetText());
                     MString s = MFileIO::loadReferenceByNode(refNode, &status);
                 }
+            }
 
-                if (!rigNamespace.empty()) {
-                    // check to see if the namespace has changed
-                    MString refNamespace = fnReference.associatedNamespace(true);
+            // Perform housekeeping on loaded references to keep them updated.
+            if (!rigNamespace.empty()) {
+                // Check to see if the namespace has changed
+                MString refNamespace = fnReference.associatedNamespace(true);
+                TF_DEBUG(PXRUSDMAYA_TRANSLATORS)
+                    .Msg(
+                        "MayaReferenceLogic::update prim=%s, namespace was: %s\n",
+                        prim.GetPath().GetText(),
+                        refNamespace.asChar());
+                if (refNamespace != rigNamespace.c_str()) {
+                    command = "file -e -ns \"";
+                    command += rigNamespace.c_str();
+                    command += "\" \"";
+                    command += filepath;
+                    command += "\"";
                     TF_DEBUG(PXRUSDMAYA_TRANSLATORS)
                         .Msg(
-                            "MayaReferenceLogic::update prim=%s, namespace was: %s\n",
+                            "MayaReferenceLogic::update prim=%s execute %s\n",
                             prim.GetPath().GetText(),
-                            refNamespace.asChar());
-                    if (refNamespace != rigNamespace.c_str()) {
-                        command = "file -e -ns \"";
-                        command += rigNamespace.c_str();
-                        command += "\" \"";
-                        command += filepath;
-                        command += "\"";
-                        TF_DEBUG(PXRUSDMAYA_TRANSLATORS)
-                            .Msg(
-                                "MayaReferenceLogic::update prim=%s execute %s\n",
-                                prim.GetPath().GetText(),
-                                command.asChar());
-                        if (!MGlobal::executeCommand(command)) {
-                            MGlobal::displayError(
-                                MString(
-                                    "Failed to update reference with new namespace. refNS:"
-                                    + refNamespace + "rigNs: " + rigNamespace.c_str() + ": ")
-                                + mayaReferencePath);
-                        }
+                            command.asChar());
+                    if (!MGlobal::executeCommand(command)) {
+                        MGlobal::displayError(
+                            MString(
+                                "Failed to update reference with new namespace. refNS:"
+                                + refNamespace + "rigNs: " + rigNamespace.c_str() + ": ")
+                            + mayaReferencePath);
```

Comment:
Note that the de-indentation for scope here is just a bonus fix: It makes it so that an outdated namespace is fixed even in the case where the file path is also being updated.